lib/segment/src/vector_storage/raw_scorer.rs

```diff
@@ -181,19 +195,17 @@ where
         }
         let mut scores = vec![];
         for point_id in points {
-            let other_vector = self.vector_storage.get_vector(point_id);
             scores.push(ScoredPointOffset {
                 idx: point_id,
-                score: TMetric::similarity(&self.query, other_vector),
+                score: self.query_scorer.score_stored(point_id),
             });
         }
         scores
     }
 
     fn check_vector(&self, point: PointOffsetType) -> bool {
-        point < self.points_count
-            // Deleted points propagate to vectors; check vector deletion for possible early return
```

Comment:
~Do we remove this bound check on purpose? IIRC we had trouble with this before, and using this bound check would give us more reliable results.~

We already do this bound check with `points_deleted`, nice!